File: API/DBAccess/Entities/Opinion.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace API.DBAccess.Entities
{
    public class Opinion
    {
        public int Id { get; set; }
        public int Rating { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public AppUser AppUser { get; set; }
        public int AppUserId { get; set; }
        public int ProductId { get; set; }
        public Product Product { get; set; }
        public bool Deleted { get; set; }
        [Required]
        public DateTime CreateDate { get; set; }
        [Required]
        public DateTime ModDate { get; set; }
    }
}

File: client/src/app/_models/opinion.ts
export class Opinion {
    id: number=0;
    rating: number;
    title: string='';
    content: string='';
    productId: number=0;
    deleted: boolean=false;
    userName: string='';
    userId: number=0;
    createDate: Date;
    modDate:Date;
    
    constructor(){}
}